yjit/src/mm.rs

```diff
@@ -0,0 +1,376 @@
+//! Memory management stuff for YJIT's code storage. Deals with virtual memory.
+// I'm aware that there is an experiment in Rust Nightly right now for to see if banning
+// usize->pointer casts is viable. It seems like a lot of work for us to participate for not much
+// benefit.
+
+use crate::utils::IntoUsize;
+
+#[cfg(not(test))]
+pub type VirtMem = VirtualMemory<sys::SystemAllocator>;
```

Comment:
Can we call it VirtualMem just so its more explicit for people new to the codebase?